[Kishore] iP

erohsikivar opened the pull request:
# NILA

>“It’s not that I’m so smart, it’s just that I stay with problems longer.” – Albert Einstein([source](https://dansilvestre.com/productivity-quotes/))

Nila is a user-friendly program designed to alleviate the burden of remembering tasks and responsibilities, making your life more organized and efficient.
 
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use!

All you need to do is,
1.download it from [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/344). 
2.double-click it. 
3.add your tasks. 
4.Nila manage your tasks 👍 
And it is **FREE**! 

## Features:
- [x] Managing tasks
- [ ] Managing deadlines
- [ ] An amazing GUI
- [ ] Reminders (coming soon)

##
If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

erohsikivar committed:
Rename, Greet, Exit

erohsikivar committed:
Echo

erohsikivar committed:
Add, List

erohsikivar committed:
Mark As Done (new class Task added)

erohsikivar committed:
ToDos, Events, Deadlines (Added to 3 subclasses for Task)

erohsikivar committed:
Automated Text UI Testing

erohsikivar committed:
Exceptions added

erohsikivar committed:
Delete tasks

erohsikivar committed:
Saving and Loading of tasks to file

erohsikivar committed:
Final branch Level-7

erohsikivar committed:
Merge branch 'branch-Level-7'

erohsikivar committed:
Date and TIme

erohsikivar committed:
Merge tag 'branch-Level-8'

erohsikivar committed:
More OOP

erohsikivar committed:
Packages

erohsikivar committed:
Merge branch 'add-gradle-support'

erohsikivar committed:
Using build gradle

erohsikivar committed:
Adding Junit

erohsikivar committed:
Added Jar

erohsikivar committed:
Added Java Doc

erohsikivar committed:
Tweaked the code to follow Coding Standard

erohsikivar committed:
Added Find method

erohsikivar committed:
Merge branch 'branch-A-JavaDoc'

erohsikivar committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Command/UserInterface.java

erohsikivar committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Command/UserInterface.java
#	src/main/java/Duke/Duke.java
#	src/main/java/Duke/Parser.java
#	src/main/java/Duke/Storage.java
#	src/main/java/Duke/TaskManager.java

erohsikivar committed:
Added Javadoc

erohsikivar committed:
A-CodingStandard

gongg21 commented on src/test/java/Duke/DukeTest.java:
```diff
@@ -0,0 +1,17 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DukeTest {
```
Is it possible to remove this class?

gongg21 commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,71 @@
+     */
+    public ArrayList<Task> load() {
+        try {
+            ArrayList<Task> taskList = new ArrayList<Task>();
+            BufferedReader reader = new BufferedReader(new FileReader(filepath));
+            String line;
+            while ((line = reader.readLine()) != null) {
```
I like your use of BufferedReader, didn't think of this!

gongg21 reviewed (commented):
Overall good coding standard!

erohsikivar committed:
Added Javadoc to push it

erohsikivar committed:
Added Javafx

erohsikivar committed:
Used Java-fxml

erohsikivar committed:
Merge branch 'branch-Java-fxml'

migfoo02 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,67 @@
+     */
+    public String getResponse(String input) {
+        return "Nila: \n" + Parser.parseCommand(input, taskManager, userInterface, storage);
+    }
```
Your code is clear and follows coding standards! However, naming your classes 'Nila' would be good to match your chatbot.

migfoo02 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,67 @@
+        while(!command.equals("bye")) {
+            Parser.parseCommand(command, taskManager, userInterface, storage);
+            command = sc.nextLine();
+        }
```
I like the condition of checking if the command is "bye" to exit the loop.

migfoo02 reviewed (commented):
Overall, your code was easy to read. Just be careful with your naming!

aliciamichellew commented on src/main/java/Command/UserInterface.java:
```diff
@@ -0,0 +1,134 @@
+     * Show welcome message.
+     */
+    public String showWelcomeMessage() {
+        return "- - - - - - - - - - - - - - - - - - - - - - - - -\n" + "Hello! I am Nila\n" +
```
Perhaps you want to use the showCommandLine() function instead of repeating this here again? Or maybe you can use constant for the long horizontal line.

aliciamichellew commented on src/main/java/Command/UserInterface.java:
```diff
@@ -0,0 +1,134 @@
+    /**
+     * To show matching results.
+     */
+
```
Nit: Maybe you can remove the empty lines here to make it neater. I noticed the same issue in several other places too

aliciamichellew commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,67 @@
+     *
+     * @param args the input arguments
+     */
+//    public static void main(String[] args) {
```
Maybe can remove the code commands that are not used?

aliciamichellew commented on src/main/java/Task/Task.java:
```diff
@@ -0,0 +1,64 @@
+    public void unmark() {
+        this.isDone = false;
+    }
+    public static Task parseFromString(String input) {
```
Any reason you do this parsing in this class instead of the Parser.java class?

aliciamichellew commented on src/test/java/Duke/TaskManagerTest.java:
```diff
@@ -0,0 +1,18 @@
+
+public class TaskManagerTest {
+    @Test
+    public void test1() {
```
Perhaps can give a better function name to know what are we testing here.

aliciamichellew commented on src/untitled/src/main/java/org/example/Main.java:
```diff
@@ -0,0 +1,7 @@
+package org.example;
+
+public class Main {
```
Not sure if I miss something here, but I think this class is not used. If it's not used, I think it will be better if you remove this.

aliciamichellew commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,148 @@
+        StringBuilder output = new StringBuilder("");
+
+        switch (action) {
+        case "todo":
```
Perhaps you would want to abstract this into several classes so that this file does not get too big and makes it easier to understand.

aliciamichellew reviewed (commented):
Overall, good work! I like how you separate your files into several packages here so that it's easier to understand.

erohsikivar committed:
Gradle : add checkstyle plugins

The checkstyle.xml file was added to detect coding style violations

erohsikivar committed:
Add assert commands

Assertions are used for improving the code reliability, I have added
assert statements to various classes and methods to enhance code quality
and assist in debugging. These assertions help ensure that critical
conditions and assumptions in the code are met during development and
testing.

-Duke class: Add assertions to validate response methods
-Parser class: Add assert statements for parsing statements.
-TaskManager class: Add assertions to validate task operations.

These assert statements will help catch unexpected issues early in the
development process, making the code more robust and easier to maintain.

erohsikivar committed:
Add Code Quality

erohsikivar committed:
Merge pull request #1 from erohsikivar/branch-A-Assertions

Add assert commands

erohsikivar committed:
Merge tag 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/Duke/Duke.java
#	src/main/java/Duke/Parser.java
#	src/main/resources/images/nila.jpg
#	src/main/resources/images/user.jpg

After resolving the conflicts, the master has implemented CodeQuality
that we learnt in class. This involves
-SLAP
-Avoid long methods
-Avoid deep nesting

This is done to ensure that the code follows standard Java coding
procedures to better understand the code. Also assert statements are
added to handle the assumptions by the programmer.

erohsikivar committed:
New feature update added

erohsikivar committed:
Changes to GUI

erohsikivar committed:
Mix minor bugs and add Ui.png

erohsikivar committed:
Add JUnit

erohsikivar committed:
Add User Guide.

erohsikivar committed:
Final Version

erohsikivar committed:
Acknowledgement of code
